client/my-sites/hosting/phpmyadmin-card/index.js

```diff
@@ -53,24 +29,32 @@ const trackOpenPhpmyadmin = () =>
 		bumpStat( 'hosting-config', 'open-phpmyadmin' )
 	);
 
-const PhpMyAdminCard = ( {
-	translate,
-	siteId,
-	token,
-	loading,
-	disabled,
-	trackOpenPhpmyadmin: trackOpenDB,
-} ) => {
-	useEffect( () => {
-		if ( token ) {
-			trackOpenDB();
-			window.open( `https://wordpress.com/pma-login?token=${ token }` );
-		}
-		return () => resetHttpData( requestId( siteId ) );
-	}, [ token, siteId ] );
+export default function PhpMyAdminCard( { disabled } ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const siteId = useSelector( getSelectedSiteId );
 
+	const [ loading, setLoading ] = useState( false );
 	const [ isRestorePasswordDialogVisible, setIsRestorePasswordDialogVisible ] = useState( false );
 
+	async function openPmaLink() {
+		setLoading( true );
+		try {
+			const { token } = await wpcom.req.post( {
+				path: `/sites/${ siteId }/hosting/pma/token`,
+				apiNamespace: 'wpcom/v2',
+			} );
+
+			if ( token ) {
+				dispatch( trackOpenPhpmyadmin() );
+				window.open( `https://wordpress.com/pma-login?token=${ token }` );
+			}
+		} catch {
+			// ignore errors
```

Comment:
I think it would be useful if we can just dispatch an error notice in this scenario.

Reply:
Notice added :+1: It's a very generic "could not open please try again" message because there's nothing that could be comprehensible or actionable for the user. The main goal is to say something instead of failing silently.